src/nvim/eval.c

```diff
@@ -5421,6 +5421,9 @@ static int list_join(garray_T *gap, list_T *l, char_u *sep, int echo_style, int
   garray_T join_ga;
   int retval;
 
+  if (l->lv_len < 1)
+    return OK;
```

Comment:
Need braces. Please put this above the declarations to make it clear `retval` and `join_ga` are unused. Lastly, please return `true` (making this return a `bool` would be welcome, but not necessary).